processing/src/main/java/org/apache/carbondata/processing/datatypes/PrimitiveDataType.java

```diff
@@ -330,21 +331,35 @@ public void writeByteArray(Object input, DataOutputStream dataOutputStream,
     }
   }
 
+  private byte[] getNullForBytes(byte[] value) {
+    String badRecordAction = CarbonProperties.getInstance()
+        .getProperty(CarbonCommonConstants.CARBON_BAD_RECORDS_ACTION);
+    if (badRecordAction.equalsIgnoreCase(CarbonCommonConstants.FORCE_BAD_RECORD_ACTION)) {
+      if (this.carbonDimension.getDataType() == DataTypes.STRING) {
+        return CarbonCommonConstants.MEMBER_DEFAULT_VAL_ARRAY;
+      } else {
+        return CarbonCommonConstants.EMPTY_BYTE_ARRAY;
+      }
+    }
+    return value;
+  }
+
   private void checkAndWriteByteArray(Object input, DataOutputStream dataOutputStream,
       BadRecordLogHolder logHolder, Boolean isWithoutConverter, String parsedValue, byte[] value)
       throws IOException {
     if (isWithoutConverter) {
       if (this.carbonDimension.getDataType() == DataTypes.STRING && input instanceof String
           && ((String)input).length() > CarbonCommonConstants.MAX_CHARS_PER_COLUMN_DEFAULT) {
-        throw new CarbonDataLoadingException("Dataload failed, String size cannot exceed "
-            + CarbonCommonConstants.MAX_CHARS_PER_COLUMN_DEFAULT + " bytes");
+        logHolder.setReason(String.format(CarbonCommonConstants.STRING_LENGTH_EXCEEDED_MESSAGE,
+            input.toString(), this.carbonDimension.getColName()));
+        value = getNullForBytes(value);
       }
       updateValueToByteStream(dataOutputStream, value);
     } else {
       if (this.carbonDimension.getDataType() == DataTypes.STRING
           && value.length > CarbonCommonConstants.MAX_CHARS_PER_COLUMN_DEFAULT) {
-        throw new CarbonDataLoadingException("Dataload failed, String size cannot exceed "
-            + CarbonCommonConstants.MAX_CHARS_PER_COLUMN_DEFAULT + " bytes");
+        logHolder.setReason(String.format(CarbonCommonConstants.STRING_LENGTH_EXCEEDED_MESSAGE,
```

Comment:
i think here after updating bad record reason, you should update null value to stream

Reply:
done.